### PR TITLE
Compile "examples" project in CI pipeline

### PR DESCRIPTION
**Pull request by rpiotrow**
* add compilation of examples project to CI pipeline
 * fix dependencies for examples project
 * improve code to eliminate compile warnings from examples project
 * bump scala version in version matrix in CI config

### CODE REVIEW

**Comment by rpiotrow** [comment]
I did not add "examples" project to aggregated projects in kebs, since there is a problem with cross building of this project. Importing `scala.jdk.CollectionConverters._` works only in 2.13 while importing `scala.collection.JavaConverters._` works in 2.12 and 2.13 but gives nasty warning in 2.13 and might not work at all in future versions of scala.

I added  additional step in CI pipeline to track if "examples" is up to date.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        scala: [2.12.12, 2.13.3]
+        scala: [2.12.12, 2.13.4]
```
Comment (dbronecki) [inline .github/workflows/ci.yml:17]: You should also update Scala version here: https://github.com/theiterators/kebs/blob/master/.mergify.yml#L15

**Comment by luksow** [comment]
Good job. Is it RTM?

**Comment by rpiotrow** [comment]
I think it is. I have updated `.mergify.yml` as @dbronecki suggested and requested him for review, but maybe he missed notification or didn't have time yet :)

**Review by dbronecki — APPROVED** [review]

**Pull request merged**